File: Easy/58. Length of Last Word/01_attempt.py
#================================
# Working process:
#   1. Iterate over s in reversed order and append it to temp list
#   2. If I walk into space, check if we have space in temp
#   3. If we do, empty the temp list
#   4. Else, return the length of temp list
#  Issue: Unnecessary list
#================================


class Solution:
    def lengthOfLastWord(self, s: str) -> int:

        temp = []
        for i in reversed(s):
            if i == ' ':
                if ' ' in temp or temp == []:
                    temp = []
                else: return len(temp)
            else: temp.append(i)

        return len(temp)

File: Easy/58. Length of Last Word/01_attempt.cpp
//================================
// Working process:
//   1. Iterate through the string s from the end to the beginning.
//   2. Keep track of the length of the last word encountered.
//   3. If a space is encountered and the length is not zero, return the length.
//   4. If a non-space character is encountered, increment the length.
// TakeAway: Learning to write code in C++ and using a simple loop to find the length of the last word in a string.
//================================


class Solution {
public:
    int lengthOfLastWord(string s) {
        int length{0};
        for(int i = s.size()-1; i >= 0; --i){
            if(s[i] == ' '){
                if(length != 0){
                    return length;
                }
            }
            else{
                ++length;
            }
        }
        return length;
    }
};